# Problem Statement 1:

In each of the following situations, state whether it is a correctly stated hypothesis
testing problem and why?
1. H0: μ = 25, H1: μ ≠ 25
2. H0: σ > 10, H1: σ = 10
3. H0: x = 50, H1: x ≠ 50
4. H0: p = 0.1, H1: p = 0.5
5. H0: s = 30, H1: s > 30

1.In this one H0 is equal to 25 and H1 is not equal to 25. so, Interpetition is correct.

2.Here , in H0 standard deviation is greater than 10 and H1 is equal to 10 but , we can't conclude to the statement is true by
standard deviation .So, the statement is not fully informed. 

3.Here , in H0 sample size is equal to 50 and H1 is not equal to 50 but , we can't conclude that the statement is true by
look into the sample size.

4.Here , in H0 p is equal to 0.1 and H1 p is  equal to 0.5.so,0.1 is the threshhold point...
H1 should be greater than 0.1 and 0.5>0.1..
So, the statement in here is true 


5.Here, in H0 sample standard deviation is 30 and H1 is greater than 30  but , we can't conclude to the statement is true by
sample standard deviation we need to know the mean atleast.So, the statement is not fully informed. 

In [9]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

# Problem Statement 2:
The college bookstore tells prospective students that the average cost of its
textbooks is Rs. 52 with a standard deviation of Rs. 4.50. A group of smart statistics
students thinks that the average cost is higher. To test the bookstore’s claim against
their alternative, the students will select a random sample of size 100. Assume that
the mean from their random sample is Rs. 52.80. Perform a hypothesis test at the
5% level of significance and state your decision.

For 5% as significance(confidence) level, we get Z(0.05) = -1.645 from z-score table

here, sample mean = 52.80, population mean = 52, sample size = 100, and std.dev. = 4.50

As , Sample error = (sample mean - population mean)/(sample size)= 0.008 = positive,so,

we will use one-tail Right Hand Test

In [2]:
def one_tailtest_right(sam_mean=1,pop_mean=1,sam_size=1,std=1,z=1.645):
    std_error = std/np.sqrt(sam_size)
    z_test = (sam_mean-pop_mean)/std_error
    if z_test > z:
        print("Null Hypothesis Rejected..")
        print("Alternative Hypothesis accepted..")
    else:
        print("Null Hypothesis accepted..")
        print("Alternative Hypothesis Rejected..")
    return round(z_test,3) 

In [3]:
one_tailtest_right(sam_mean=52.80,pop_mean=52,sam_size=100,std=4.50,z=-1.645)

Null Hypothesis Rejected..
Alternative Hypothesis accepted..


1.778

# Problem Statement 3:
A certain chemical pollutant in the Genesee River has been constant for several
years with mean μ = 34 ppm (parts per million) and standard deviation σ = 8 ppm. A
group of factory representatives whose companies discharge liquids into the river is
now claiming that they have lowered the average with improved filtration devices. A
group of environmentalists will test to see if this is true at the 1% level of
significance. Assume \ that their sample of size 50 gives a mean of 32.5 ppm.
Perform a hypothesis test at the 1% level of significance and state your decision.

For 1% as significance(confidence) level, we get Z(0.01) = -2.3 from z-score table

here, sample mean = 32.5, population mean = 34, sample size = 50, and std.dev. = 8

As , Sample error = (sample mean - population mean)/(sample size)= -0.03 = Negative,so,

we will use one-tail Left Hand Test

In [7]:
def one_tailtest_left(sam_mean=1,pop_mean=1,sam_size=1,std=1,z=-1.645):
    std_error = std/np.sqrt(sam_size)
    z_test = (sam_mean-pop_mean)/std_error
    if z_test < z  :
        print("Null Hypothesis Rejected..")
        print("Alternative Hypothesis accepted..")
    else:
        print("Null Hypothesis accepted..")
        print("Alternative Hypothesis Rejected..")   
    return round(z_test,3) 

In [8]:
one_tailtest_left(sam_mean=32.5,pop_mean=34,sam_size=50,std=8,z=-2.3)

Null Hypothesis accepted..
Alternative Hypothesis Rejected..


-1.326

# Problem Statement 4:
Based on population figures and other general information on the U.S. population,
suppose it has been estimated that, on average, a family of four in the U.S. spends
about $1135 annually on dental expenditures. Suppose further that a regional dental
association wants to test to determine if this figure is accurate for their area of
country. To test this, 22 families of 4 are randomly selected from the population in
that area of the country and a log is kept of the family’s dental expenditure for one
year. The resulting data are given below. Assuming, that dental expenditure is
normally distributed in the population, use the data and an alpha of 0.5 to test the
dental association’s hypothesis.

1008, 812, 1117, 1323, 1308, 1415, 831, 1021, 1287, 851, 930, 730, 699,
872, 913, 944, 954, 987, 1695, 995, 1003, 994

In [9]:
l=[1008, 812, 1117, 1323, 1308, 1415, 831, 1021, 1287, 851, 930, 730, 699, 872, 913, 944, 954, 987, 1695, 995, 1003, 994]
S = sum(l)
print(S)
print (round((S/22),2))

22689
1031.32


In [10]:
std=round(np.sqrt(((S-1135)**2)/(22-1)),2)
std

4703.47

Here , population mean=1135 , sample mean =1031 , alpha = 0.5, degree_of_freedom(df)=(n-4)=19

According to t-table , t(0.5,19) = 1.729 , sample size = 22 , standard deviation = 4703

As , Sample error = (sample mean - population mean)/(sample size)= -4.72 = Negative,so,

we will use one-tail Left Hand Test

In [11]:
def one_tailtest_left(sam_mean=1,pop_mean=1,sam_size=1,std=1,z=-1.645):
    std_error = std/np.sqrt(sam_size)
    z_test = (sam_mean-pop_mean)/std_error
    if z_test < z  :
        print("Null Hypothesis Rejected..")
        print("Alternative Hypothesis accepted..")
    else:
        print("Null Hypothesis accepted..")
        print("Alternative Hypothesis Rejected..")   
    return round(z_test,3) 

In [12]:
one_tailtest_left(sam_mean=1031,pop_mean=11135,sam_size=22,std=4703,z=1.729)

Null Hypothesis Rejected..
Alternative Hypothesis accepted..


-10.077

# Problem Statement 5:
In a report prepared by the Economic Research Department of a major bank the
Department manager maintains that the average annual family income on Metropolis
is $48,432. 

What do you conclude about the validity of the report if a random sample
of 400 families shows and average income of $48,574 with a standard deviation of
2000?

In [13]:
SE=(48574-48432)/400
SE

0.355

Assume , significance level =5 % .z(0.05) = -1.645  

here , sample mean = 48574 , population mean = 48432 ,standard deviation = 2000 , sample size = 400

Sample error = (sample mean - population mean)/(sample size)= 0.355 = positive,so,

we will use one-tail Right Hand Test

H0 population mean = 48432

H1 population mean >48432

In [14]:
def one_tailtest_right(sam_mean=1,pop_mean=1,sam_size=1,std=1,z=1.645):
    std_error = std/np.sqrt(sam_size)
    z_test = (sam_mean-pop_mean)/std_error
    if z_test > z:
        print("Null Hypothesis Rejected..")
        print("Alternative Hypothesis accepted..")
    else:
        print("Null Hypothesis accepted..")
        print("Alternative Hypothesis Rejected..")
    return round(z_test,3) 

In [15]:
one_tailtest_right(sam_mean=48574,pop_mean=48432,sam_size=400,std=2000,z=-1.645)

Null Hypothesis Rejected..
Alternative Hypothesis accepted..


1.42

So , Reports are vallied.

# Problem Statement 6:
Suppose that in past years the average price per square foot for warehouses in the
United States has been $32.28. A national real estate investor wants to determine

whether that figure has changed now. The investor hires a researcher who randomly

samples 19 warehouses that are for sale across the United States and finds that the

mean price per square foot is $31.67, with a standard deviation of 

$1.29. assume
that the prices of warehouse footage are normally distributed in population. If the
researcher uses a 5% level of significance, what statistical conclusion can be
reached? What are the hypotheses?

In [16]:
SE = (31.67-32.28)/19
SE

-0.032105263157894706

Here , sample mean = 31.67 , population mean = 32.28 , sample size = 19 , standard deviation = 1.29

At the 5% significance level z(0.05) = -1.64

As, sample error = (sample mean - population mean)/sample size = -0.032 = Negative , we will use

one tail Left Hand test.

In [17]:
def one_tailtest_left(sam_mean=1,pop_mean=1,sam_size=1,std=1,z=-1.645):
    std_error = std/np.sqrt(sam_size)
    z_test = (sam_mean-pop_mean)/std_error
    if z_test < z  :
        print("Null Hypothesis Rejected..")
        print("Alternative Hypothesis accepted..")
    else:
        print("Null Hypothesis accepted..")
        print("Alternative Hypothesis Rejected..")   
    return round(z_test,3) 

In [18]:
one_tailtest_left(sam_mean=31.67,pop_mean=32.28,sam_size=19,std=1.29,z=-1.645)

Null Hypothesis Rejected..
Alternative Hypothesis accepted..


-2.061

# Problem Statement 8:
Find the t-score for a sample size of 16 taken from a population with mean 10 when
the sample mean is 12 and the sample standard deviation is 1.5.


In [19]:
SE=(12-10)/16
SE

0.125

Here , sample mean = 12 , population mean = 10 , sample size = 16 , standard deviation = 1.5

Let's assume , significance level is 5%. z(0.05) = -1.64

As, sample error = (sample mean - population mean)/sample size = 0.125 = positive , we will use

one tail Right Hand test.

In [20]:
def one_tailtest_right(sam_mean=1,pop_mean=1,sam_size=1,std=1,z=1.645):
    std_error = std/np.sqrt(sam_size)
    z_test = (sam_mean-pop_mean)/std_error
    if z_test > z:
        print("Null Hypothesis Rejected..")
        print("Alternative Hypothesis accepted..")
    else:
        print("Null Hypothesis accepted..")
        print("Alternative Hypothesis Rejected..")
    return round(z_test,3) 

In [21]:
one_tailtest_right(sam_mean=12,pop_mean=10,sam_size=16,std=1.5,z=-1.645)

Null Hypothesis Rejected..
Alternative Hypothesis accepted..


5.333

# Problem Statement 9:
Find the t-score below which we can expect 99% of sample means will fall if samples
of size 16 are taken from a normally distributed population. 

Here , (1-alpha) = 99% = 0.99 , alpha = (1-0.99)=0.01 , degree of freedom = (16-1) = 15 
According to t-table, t(0.99/15)=-t(0.01/15)=-2.947 , sample size = 16 


In [29]:
sample_size = 16
alpha = (1-(99/100))
degree_of_freedom = (sample_size-1)
degree_of_freedom 
#According to t-table,t(0.99/15)=-t(0.01/15)=-2.947
t=-2.947
t

-2.947

# Problem Statement 10:
If a random sample of size 25 drawn from a normal population gives a mean of 60
and a standard deviation of 4, find the range of t-scores where we can expect to find
the middle 95% of all sample means. Compute the probability that (−𝑡0.05 <𝑡<𝑡0.10).

In [24]:
sam_size=25 
std=4 
t1=0.05 
SE=std/np.sqrt(25)  
zx1=(-0.05-0.95)/SE #According to z table
print(zx1)
z1= 0.1056 
zx2=(0.10-0.95)/SE  #According to z table 
print(zx2)
z2= 0.1446 
z1,z2

-1.25
-1.0625


(0.1056, 0.1446)

In [25]:
def two_tail_ttest(sam_mean=1,pop_mean=1,sam_size=1,std=1,t1=1,t2=1):
    #sam_mean = pop_mean + 2*(std/(sam_size**0.5))
    std_error = std/np.sqrt(sam_size)
    t_test = (sam_mean-pop_mean)/std_error
    if (t1 < t_test < t2) :
        print("Null Hypothesis is accepted...")
        print("Alternate Hypothesis is rejected...")
    else:
        print("Null Hypothesis is rejected...")
        print("Alternate Hypothesis is accepted...")
    return round(t_test,3)  

In [26]:
two_tail_ttest(sam_mean=61.6,pop_mean=60,sam_size=25,std=4,t1=0.1056,t2=0.1446)

Null Hypothesis is rejected...
Alternate Hypothesis is accepted...


2.0

# Problem Statement 11:
Two-tailed test for difference between two population means
Is there evidence to conclude that the number of people travelling from Bangalore to
Chennai is different from the number of people travelling from Bangalore to Hosur in
a week, given the following:

Population 1: Bangalore to Chennai n1 = 1200

x1 = 452

s1 = 212

Population 2: Bangalore to Hosur n2 = 800

x2 = 523

s2 = 185

Here, n > 30 where, n1 = 1200, n2 = 800 ; X1 = 452, x2 = 523  ; s1 = 212 , s2 = 185 
Let's consider , significance_level = 5% ,alpha/2 = 2.5% as it is two-tail-test..
So, Z(0.025) = 1.96

In [18]:
def similarity_testa(n1=1, n2=1, X1=1, x2=1, s1=1, s2=1,z=1):
    serror = np.sqrt((s1**2/n1)+(s2**2/n2))
    t_test = (X1-x2)/serror
    if (-z < t_test < z):
        print("Null Hypothesis is accepted...")
        print("So, Both the populations are similar...")
    else:
        print("Null Hypothesis is rejected...")
        print("So, the populations are not similar...")
    return round(t_test,3) 

In [19]:
similarity_testa(n1=1200, n2=800, X1=452, x2=523, s1=212, s2=185,z=1.96)

Null Hypothesis is rejected...
So, the populations are not similar...


-7.926

# Problem Statement 12:
Is there evidence to conclude that the number of people preferring Duracell battery is
different from the number of people preferring Energizer battery, given the following:
Population 1: Duracell

n1 = 100

x1 = 308
s1 = 84

Population 2: Energizer

n2 = 100

x2 = 254
s2 = 67


Let's consider , significance_level = 5% ,alpha/2 = 2.5% as it is two-tail-test.. So, Z(0.025) = 1.96

In [20]:
def similarity_testa(n1=1, n2=1, X1=1, x2=1, s1=1, s2=1,z=1):
    serror = np.sqrt((s1**2/n1)+(s2**2/n2))
    t_test = (X1-x2)/serror
    if (-z < t_test < z):
        print("Null Hypothesis is accepted...")
        print("So, Both the populations are similar...")
    else:
        print("Null Hypothesis is rejected...")
        print("So, the populations are not similar...")
    return round(t_test,3) 

In [21]:
similarity_testa(n1=100, n2=100, X1=308, x2=254, s1=84, s2=67,z=1.96)

Null Hypothesis is rejected...
So, the populations are not similar...


5.026

# Problem Statement 13:
Pooled estimate of the population variance
Does the data provide sufficient evidence to conclude that average percentage
increase in the price of sugar differs when it is sold at two different prices?

Population 1: Price of sugar = Rs. 27.50

n1 = 14

x1 = 0.317%

s1 = 0.12%

Population 2: Price of sugar = Rs. 20.00

n2 = 9

x2 = 0.21%

s2 = 0.11%

In [48]:
x1=round(27.50*(0.317/100),3)
x2=round(20*(0.21/100),3)
s1=round(27.50*(0.12/100),4)
s2=round(20*(0.11/100),4)
print('x1=',x1)
print('x2=',x2)
print('s1=',s1)
print('s2=',s2)
print('s1^2=',round((s1)**2,5))
print('s2^2=',round((s2)**2,5))

x1= 0.087
x2= 0.042
s1= 0.033
s2= 0.022
s1^2= 0.00109
s2^2= 0.00048


**Here no. of data sets n < 30  n1 = 14, n2 = 9, x1 = 0.087, x2 = 0.042, s1^2 = 0.00109, s2^2 = 0.00048,

SD = sqrt[(s1^2+s2^2)/(nx+ny-2)], significance = 5% for two-tail-test, 

deg. of freedom => (14+9-2) = 21, t(0.025)=2.080 from t-table.

Let's consider , significance_level = 5% ,alpha/2 = 2.5% as it is two-tail-test.. So, Z(0.025) = 1.96

In [51]:
def similarity_test(n1=1, n2=1, x1=1, x2=1, s1=1, s2=1,z=1):
    sd = np.sqrt((s1**2+s2**2)/(n1+n2-2))
    serror = sd*np.sqrt((1/n1)+(1/n2))
    serror = np.sqrt((s1**2/n1)+(s2**2/n2))
    t_test = (x1-x2)/serror
    if (-z < t_test < z):
        print("Null Hypothesis is accepted...")
        print("So, Both the populations are similar...")
    else:
        print("Null Hypothesis is rejected...")
        print("So, the populations are not similar...")
    return round(t_test,3) 

In [52]:
t = 2.080
t_test = similarity_test(n1=14,n2=9,x1=0.087,x2=0.042,s1=0.033,s2=0.022,z=t)
t_test

Null Hypothesis is rejected...
So, the populations are not similar...


3.923

# Problem Statement 14:
The manufacturers of compact disk players want to test whether a small price
reduction is enough to increase sales of their product. Is there evidence that the
small price reduction is enough to increase sales of compact disk players?
Population 1: Before reduction
n1 = 15
x1 = Rs. 6598 s1 = Rs. 844
Population 2: After reduction n2 = 12
x2 = RS. 6870
s2 = Rs. 669


Let's consider , significance_level = 5% ,alpha/2 = 2.5% as it is two-tail-test.. So, Z(0.025) = 1.96

In [22]:
def similarity_testa(n1=1, n2=1, X1=1, x2=1, s1=1, s2=1,z=1):
    serror = np.sqrt((s1**2/n1)+(s2**2/n2))
    t_test = (X1-x2)/serror
    if (-z < t_test < z):
        print("Null Hypothesis is accepted...")
        print("So, Both the populations are similar...")
    else:
        print("Null Hypothesis is rejected...")
        print("So, the populations are not similar...")
    return round(t_test,3) 

In [23]:
similarity_testa(n1=15, n2=12, X1=6598, x2=6870, s1=844, s2=669,z=1.96)

Null Hypothesis is accepted...
So, Both the populations are similar...


-0.934

# Problem Statement 15:
Comparisons of two population proportions when the hypothesized difference is zero
Carry out a two-tailed test of the equality of banks’ share of the car loan market in
1980 and 1995.
Population 1: 1980

n1 = 1000

x1 = 53

𝑝^1 = 0.53
Population 2: 1985

n2 = 100

x2 = 43

𝑝^2= 0.53

Let's consider , significance_level = 5% ,alpha/2 = 2.5% as it is two-tail-test.. So, Z(0.025) = 1.96

In [53]:
def similarity_testa(n1=1, n2=1, x1=1, x2=1, s1=1, s2=1,z=1):
    serror = np.sqrt((s1**2/n1)+(s2**2/n2))
    t_test = (x1-x2)/serror
    if (-z < t_test < z):
        print("Null Hypothesis is accepted...")
        print("So, Both the populations are similar...")
    else:
        print("Null Hypothesis is rejected...")
        print("So, the populations are not similar...")
    return round(t_test,3) 

In [54]:
similarity_testa(n1=1000, n2=100, x1=53, x2=43, s1=0.53, s2=0.53,z=1.96)

Null Hypothesis is rejected...
So, the populations are not similar...


179.899

# Problem Statement 16:
Carry out a one-tailed test to determine whether the population proportion of
traveler’s check buyers who buy at least $2500 in checks when sweepstakes prizes
are offered as at least 10% 

higher than the proportion of such buyers when no sweepstakes are on.

Population 1: With sweepstakes

n1 = 300

x1 = 120

𝑝^1 = 0.40

Population 2: No sweepstakes n2 = 700

x2 = 140
p^2= 0.20

Let's consider , significance_level = 5% ,alpha/2 = 2.5% as it is two-tail-test.. So, Z(0.025) = 1.96

In [58]:
def similarity_testa(n1=1, n2=1, x1=1, x2=1, s1=1, s2=1,z=1):
    serror = np.sqrt((s1**2/n1)+(s2**2/n2))
    t_test = (x1-x2)/serror
    if (-z < t_test < z):
        print("Null Hypothesis is accepted...")
        print("So, Both the populations are similar...")
    else:
        print("Null Hypothesis is rejected...")
        print("So, the populations are not similar...")
    return round(t_test,3) 

In [59]:
similarity_testa(n1=300, n2=700, x1=120, x2=140, s1=0.40, s2=0.20,z=1.96)

Null Hypothesis is rejected...
So, the populations are not similar...


-823.055

# Problem Statement 17:
A die is thrown 132 times with the following results: Number turned up: 1, 2, 3, 4, 5, 6
Frequency: 16, 20, 25, 14, 29, 28
Is the die unbiased? Consider the degrees of freedom as 𝑝^-1 . 

On the hypothesis that the die is unbiased we should expect the frequency of each number to be 132/6 =22

so, O=[16 ,20 ,25 ,14 ,29 ,28 ]  & E=[22 ,22 ,22 ,22 ,22 ,22 ]

Let's consider significance level = 5% , alpha=0.05.
Degree of freedom = p^-1 = 6-1 =5
So, chi(0.05) = 11.070 from chi-table

In [71]:
O=[ 16, 20, 25, 14, 29, 28] 
E= [22, 22, 22, 22, 22, 22 ]

def biased_check(O,E,ch=1):
    chi=0
    for i in range(len(O)):
        chi = chi + (O[i]-E[i])**2/E[i]
    if chi < ch:
        print("Null Hypothesis is accepted... ")
        print("so,The die is unbiased..")
    else:
        print("Null Hypothesis is rejected...")
        print("so,The die is biased....")
    return round(chi,3)  

In [72]:
biased_check(O,E,ch=11.070)

Null Hypothesis is accepted... 
so,The die is unbiased..


9.0

# Problem Statement 18:
In a certain town, there are about one million eligible voters. A simple random
sample of 10,000 eligible voters was chosen to study the relationship between
gender and participation in the last election. The results are summarized in the
following 2X2 (read two by two) contingency table:

                                  
voted ----- Men = 2792_______Women=3591

Not voted---------Men= 1486____Women=2131

We would want to check whether being a man or a woman (columns) is independent of
having voted in the last election (rows). In other words, is “gender and voting independent”?

In [84]:
#The observed table with totals included is:
M_voted=2792
w_voted=3591
Tot_voted=6383
M_notvoted=1486
w_notvoted=2131
Tot_notvoted=3617
M_Tot=4278
w_Tot=5722
Total=10000
Expected_value_Mvoted=(Tot_voted*M_Tot)/Total
Expected_value_Wvoted=(Tot_voted*w_Tot)/Total
print(round((Expected_value_Mvoted),0))
print(round((Expected_value_Wvoted),0))
Expected_value_Mnvoted=(Tot_notvoted*M_Tot)/Total
Expected_value_Wnvoted=(Tot_notvoted*w_Tot)/Total
print(round((Expected_value_Mnvoted),0))
print(round((Expected_value_Wnvoted),0))

2731.0
3652.0
1547.0
2070.0


In [81]:
#The Expectation table with totals included is:
M_voted=2731
w_voted=3652
Tot_voted=(2731+3652)
M_notvoted=1547
w_notvoted=2070
Tot_notvoted=(1547+2070)
Tot_voted,Tot_notvoted

(6383, 3617)

Here, o=[2792 , 3591 , 1486 ,2131]
E=[2731 , 3652 , 1547 ,2070] and degree of freedom will be (2-1)*(2-1)=1

Let's Assume, significance level = 5% , alpha=0.05

so, according to chi table, chi(0.05)=3.841

In [91]:
O=[2792 , 3591 , 1486 ,2131]
E=[2731 , 3652 , 1547 ,2070]
def voting_check(O,E,ch=1):
    chi=0
    for i in range(len(O)):
        chi = chi + (O[i]-E[i])**2/E[i]
    if chi < ch:
        print("Null Hypothesis is accepted...")
        print("So, gender and voting are independent")
    else:
        print("Null Hypothesis is rejected...")
        print("So, gender and voting are dependent")
    return round(chi,3)  

In [92]:
voting_check(O,E,ch=3.841)

Null Hypothesis is rejected...
So, gender and voting are dependent


6.584

# Problem Statement 19:
A sample of 100 voters are asked which of four candidates they would vote for in an
election. The number supporting each candidate is given below:

Higgins = 41   Reardon = 19   white = 24    charlton = 16  


Do the data suggest that all candidates are equally popular? [Chi-Square = 14.96,
with 3 df, 𝑝^<0.05 ].

Here, we should expect the frequency of each number to be 100/4 =25

O = [41 , 19 ,24 ,16] E = [25 , 25 ,25 ,25]

In [96]:
O = [41 , 19 ,24 ,16] 
E = [25 , 25 ,25 ,25]
def popularity_check(O,E,ch=1):
    chi=0
    for i in range(len(O)):
        chi = chi + (O[i]-E[i])**2/E[i]
    if chi < ch:
        print("Null Hypothesis is accepted... ")
        print("so,all candidates are equally popular..")
    else:
        print("Null Hypothesis is rejected...")
        print("so,all candidates are not equally popular")
    return round(chi,3)

In [97]:
popularity_check(O,E,ch=14.96)

Null Hypothesis is accepted... 
so,all candidates are equally popular..


14.96

#    Problem Statement 20:
Children of three ages are asked to indicate their preference for three photographs of
adults. Do the data suggest that there is a significant relationship between age and
photograph preference? What is wrong with this study? [Chi-Square = 29.6, with 4
df: 𝑝 < 0.05]

                                               
Age of child: 5-6 years:          
A = 18

B = 22

C = 20

Age of child: 7-8 years:          
A = 2

B = 28 

C = 40


Age of child: 9-10 years:          
A = 20

B = 10

C = 40                               

Here , E = (row_total*column_total)/grand_total

Age of child: 5-6 years:

AE=(18+2+20)*18/(18+22+20)=12

BE=(22+28+10)*18/(18+22+20)=18

CE=(20+40+40)*18/(18+22+20)=30

Age of child: 7-8 years:

AE=(18+2+20)*26/(28+2+40)=14

BE=(22+28+10)*25/(28+2+40)=21

CE=(20+40+40)*24/(28+2+40)=35

Age of child: 9-10 years:

AE=(18+2+20)*26/(20+10+40)=14

BE=(22+28+10)*25/(28+2+40)=21

CE=(20+40+40)*24/(28+2+40)=35

According to that we get , E =[12, 18 , 30  ,14, 21, 35, 14, 21, 35 ]

O=[18 , 22 ,20 ,2 ,28 ,40 ,20 ,10 ,40]

In [98]:
O=[18 , 22 ,20 ,2 ,28 ,40 ,20 ,10 ,40]
E =[12, 18 , 30  ,14, 21, 35, 14, 21, 35 ]

def relation(O,E,ch=1):
    chi=0
    for i in range(len(O)):
        chi = chi + (O[i]-E[i])**2/E[i]
    if chi < ch:
        print("Null Hypothesis is accepted... ")
        print("so,there is no significant relationship between the age of child and the photographer preference..")
    else:
        print("Null Hypothesis is rejected...")
        print("so,there is  significant relationship between the age of child and the photographer preference..")
    return round(chi,3)

In [99]:
relation(O,E,ch=29.6)

Null Hypothesis is rejected...
so,there is  significant relationship between the age of child and the photographer preference..


29.603

# Problem Statement 21:
A study of conformity using the Asch paradigm involved two conditions: one where
one confederate supported the true judgement and another where no confederate
gave the correct response.

conform : 

support = 18

No support = 40

No conform:

support = 32

No support = 10

Is there a significant difference between the "support" and "no support" conditions in the
frequency with which individuals are likely to conform? [Chi-Square = 19.87, with 1 df:
𝑝 < 0.05].


It is from famous Asche conformity studies in 1920,so here,
O=[ 18 , 40 , 32 , 10 ]

E = [29 , 29 , 21 , 21 ]

In [1]:
O=[ 18 , 40 , 32 , 10 ]
E = [29 , 29 , 21 , 21 ]

def Asche_conformity(O,E,ch=1):
    chi=0
    for i in range(len(O)):
        chi = chi + (O[i]-E[i])**2/E[i]
    if chi < ch:
        print("Null Hypothesis is accepted... ")
        print("so,there is a significant difference" )
    else:
        print("Null Hypothesis is rejected...")
        print("so,there is not any significant difference" )
    return round(chi,3)  

In [2]:
Asche_conformity(O,E,ch=19.87)

Null Hypothesis is accepted... 
so,there is a significant difference


19.869

# Problem Statement 22:
We want to test whether short people differ with respect to their leadership qualities
(Genghis Khan, Adolf Hitler and Napoleon were all stature-deprived, and how many midget
MP's are there?) The following table shows the frequencies with which 43 short people and
52 tall people were categorized as "leaders", "followers" or as "unclassifiable". Is there a
relationship between height and leadership qualities?
[Chi-Square = 10.71, with 2 df: 𝑝 < 0.01].

Short:Leader=12	

Tall:Leader = 32

Follower : short =22

Tall = 14

Unclassified: short = 9	

Tall = 6

In [4]:
O=[12 ,32 ,22 ,14 ,9 ,6]
E=[19.92 , 24.08 ,16.29 , 19.71 ,6.76 , 8.21 ]
def relation(O,E,ch=1):
    chi=0
    for i in range(len(O)):
        chi = chi + (O[i]-E[i])**2/E[i]
    if chi < ch:
        print("Null Hypothesis is accepted... ")
        print("so,there is a relationship between height and leadership qualities." )
    else:
        print("Null Hypothesis is rejected...")
        print("o,there is a no relationship between height and leadership qualities." )
    return round(chi,3)  

In [5]:
relation(O,E,ch=10.71)

Null Hypothesis is rejected...
o,there is a no relationship between height and leadership qualities.


10.747

# Problem Statement 23:
Each respondent in the Current Population Survey of March 1993 was classified as
employed, unemployed, or outside the labor force. The results for men in California age 35-
44 can be cross-tabulated by marital status, as follows:
            
Employed :---- 

Married = 679  

Widowed, divorced or separated = 103  

Never married =114

Unemployed:-----  

Married = 63 

Widowed, divorced or separated = 10 

Never married = 20

Not in labor force :----

Married = 42          

Widowed, divorced or separated = 18               

Never married = 25

Men of different marital status seem to have different distributions of labor force status. Or is
this just chance variation? (you may assume the table results from a simple random
sample.)


In [8]:
dict = {'Married':[679,63,42],'Wid_div_sep':[103,10,18],'Never_married':[114,20,25]}

In [11]:
df = pd.DataFrame(dict,index=['Employed','Unemployed','Not_in_labor_Force'])
df

Married  Wid_div_sep  Never_married
Employed                679          103            114
Unemployed               63           10             20
Not_in_labor_Force       42           18             25

In [23]:
Mm = df.Married.mean()
Wm = df.Wid_div_sep.mean()
Nm = df.Never_married.mean()
Em = df.loc['Employed'].mean()
Um = df.loc['Unemployed'].mean()
NLFm = df.loc['Not_in_labor_Force'].mean()
lst1 = [Mm,Wm,Nm]
lst2 = [Em,Um,NLFm]

In [45]:
df['Mean']=lst2
df.columns

Index(['Married', 'Wid_div_sep', 'Never_married', 'Mean'], dtype='object')

In [55]:
#dict2 = {'Married', 'Wid_div_sep', 'Never_married', 'Mean'}
df1 = pd.DataFrame(lst1,columns=['mean'],index=['Married', 'Wid_div_sep', 'Never_married']).T
df1

Married  Wid_div_sep  Never_married
mean  261.333333    43.666667           53.0

In [58]:
df3 = pd.concat([df,df1],0)

In [59]:
df3

Married  Wid_div_sep  Never_married        Mean
Employed            679.000000   103.000000          114.0  298.666667
Unemployed           63.000000    10.000000           20.0   31.000000
Not_in_labor_Force   42.000000    18.000000           25.0   28.333333
mean                261.333333    43.666667           53.0         NaN

So, degree of freedom =(3-1)*(3-1)=4

chi(0.05 ,4)=9.487729

In [63]:
O=[679 , 103 , 114 ,63 ,10 ,20 ,42 ,18 ,25]
E=[654.06 , 109.29 , 132.65 , 67.89 ,   11.34 , 13.77 , 62.05 , 10.37 , 12.58 ]
def survey(O,E,ch=1):
    chi=0
    for i in range(len(O)):
        chi = chi + (O[i]-E[i])**2/E[i]
    if chi < ch:
        print("Null Hypothesis is accepted... ")
        print("so,there is a relationship between rows and columns." )
    else:
        print("Null Hypothesis is rejected...")
        print("so,there is a no relationship between rows and columns." )
    return round(chi,3)

In [64]:
 survey(O,E,ch=9.487729)

Null Hypothesis is rejected...
so,there is a no relationship between rows and columns.


31.619